### ISSUE TITLE
Prettier with long list of optional named parameters (using destructuring) not so pretty

### ISSUE THREAD

**Issue by kasperpeulen** [prettier/prettier#3016]
I have this class:

```javascript
class Model {
  /**
   * @param id {String}
   * @param createdAt {Date}
   * @param updatedAt {Date}
   * @param endedAt {Date}
   */
  constructor({
      id = uuid(),
      createdAt = new Date(),
      updatedAt = new Date(),
      endedAt = null,
  } = {}) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.endedAt = endedAt;
  }
}
```

Prettier makes this from it:

```javascript
class Model {
  /**
   * @param id {String}
   * @param createdAt {Date}
   * @param updatedAt {Date}
   * @param endedAt {Date}
   */
  constructor(
    {
      id = uuid(),
      createdAt = new Date(),
      updatedAt = new Date(),
      endedAt = null,
    } = {},
  ) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.endedAt = endedAt;
  }
}
```

Which feels a bit weird to me. If I remove the default value of `{}`, it becomes much prettier:

```javascript
class Model {
  /**
   * @param id {String}
   * @param createdAt {Date}
   * @param updatedAt {Date}
   * @param endedAt {Date}
   */
  constructor({
    id = uuid(),
    createdAt = new Date(),
    updatedAt = new Date(),
    endedAt = null,
  }) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.endedAt = endedAt;
  }
}
```

But now I need to always explicitly give the first argument, as `new Model()` gives now an error (wished Javascript behaved differently). I have to write `new Model({})` in that case, if I want the default values.

**Comment by duailibe** [comment]
Fixed in 1.9.2!

**Comment by kasperpeulen** [comment]
Thanks 🙏🙏
